back-end/lms/views/absence.py

```diff
@@ -116,44 +162,122 @@ def put(self, request: Request) -> Response:
-        except IntegrityError:
+        return Response(
+            {
+                'message': f'Absence with id {request.data["id"]} '
+                           f'does not exist in this database'
+            },
+            status=HTTP_400_BAD_REQUEST)
+    
+    # pylint: disable=no-self-use
+    @csrf_exempt
+    def delete(self, request: Request) -> Response:
+        """
+        DELETE - function uses id from request 'query'
+        :param request:
+        :return:
+        """
+        absence_to_delete = Absence.objects.filter(
+            id=request.query_params['id'])
+        if absence_to_delete.exists():
+            absence_to_delete.delete()
             return Response(
                 {
-                    'message': 'A record with this student and this date '
-                               'already exists. Please, modify existing record.'
+                    'message': f'Absence with id {request.query_params["id"]} '
+                               f'successfully deleted'
                 },
-                status=HTTP_400_BAD_REQUEST)
+                status=HTTP_200_OK)
+        return Response(
+            {
+                'message': f'Absence with id {request.query_params["id"]} '
+                           f'does not exist in this database'
+            },
+            status=HTTP_400_BAD_REQUEST)
 
 
 @permission_classes((AllowAny,))
-class AbsenceJournal(APIView):
+class AbsenceJournalView(APIView):
 
     @csrf_exempt
     def get(self, request: Request) -> Response:
         """
         Get absent records in the form of journal
         GET syntax example:
-        .../absence_journal/?milgroup=1809
+        .../absence_journal/?milgroup=1809&datefrom=...&dateto=...
         :param request:
         :return:
         """
-        if 'milgroup' not in request.query_params:
-            return Response(
-                {'message': 'Please, insert milgroup as a query parameter.'},
-                status=HTTP_400_BAD_REQUEST)
+        required_params = ['milgroup', 'date_from', 'date_to']
+        for req_param in required_params:
+            if req_param not in request.query_params:
+                return Response(
+                    {'message': f'Please, insert {req_param} as a query parameter.'},
+                    status=HTTP_400_BAD_REQUEST)
+        
 
         absences = Absence.objects.filter(
             studentid__milgroup__milgroup=request.query_params['milgroup'])
-        return Response({'absences': absences.data}, status=HTTP_200_OK)
+        
+        data = {}
+
+        # add milgroup data
+        milgroup = MilgroupSerializer(Milgroup.objects.get(
+            milgroup=request.query_params['milgroup'])).data
+        data['milgroup'] = milgroup
+
+        # calculate dates
+        date_from = datetime.strptime(request.query_params['date_from'], '%d.%m.%Y')
+        date_to = datetime.strptime(request.query_params['date_to'], '%d.%m.%Y')
```

Comment:
What's the story with these date formats anyway?

Reply:
I am trying to stick with the Russian Format as much as possible.